src/rgw/rgw_acl.h

```diff
@@ -214,57 +217,30 @@ struct ACLReferer {
       perm(perm) {
   }
 
-  bool is_match(std::string http_referer) const {
-    std::string http_host;
-    if (!get_http_host(http_referer, http_host))
+  bool is_match(boost::string_ref http_referer) const {
+    boost::optional<boost::string_ref> http_host = get_http_host(http_referer);
```

Comment:
Hmm, maybe `const` here?
BTW: using `auto` is also perfectly fine in the project.